### PR TITLE
[ASA-237][JR] Added authentication to the /:nino/:year/all-details endpoint

### PR DESCRIPTION
**Pull request by johnrevill**

### CODE REVIEW

File: test/uk/gov/hmrc/taxhistory/connectors/CitizenDetailsConnectorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  private val testConnector = new CitizenDetailsConnector(http = mockHttp, baseUrl = "/test", metrics = mockMetrics)
+
+  override def beforeEach = {
+    reset(mockHttp)
+    reset(mockMetrics)
+    reset(mockTimerContext)
+  }
+
+  "lookupSaUtr" should {
```
Comment (arkowalski) [inline test/uk/gov/hmrc/taxhistory/connectors/CitizenDetailsConnectorSpec.scala]: lets have verify metrics on all cases

File: test/uk/gov/hmrc/taxhistory/controllers/PayAsYouEarnControllerSpec.scala [hunk 2]
```diff
@@ -17,31 +17,31 @@
 package uk.gov.hmrc.taxhistory.controllers
 
 import org.joda.time.LocalDate
-import org.mockito.Matchers._
+import org.mockito.Matchers.{any, eq => meq}
```
Comment (arkowalski) [inline test/uk/gov/hmrc/taxhistory/controllers/PayAsYouEarnControllerSpec.scala]: use original names?

Comment (johnrevill) [reply test/uk/gov/hmrc/taxhistory/controllers/PayAsYouEarnControllerSpec.scala]: Original name from the Java library is `eq` which conflicts with Scala's completely different `eq`, hence the rename.

File: test/uk/gov/hmrc/taxhistory/services/SaAuthServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        override def authConnector = mockAuthConnector
+      }
+
+      when(mockSaAuthPredicateBuilder.authorisationPredicate(meq(validNino))(any()))
+        .thenReturn(Future.successful(mockPredicate))
+    }
+
+    "invoke code block and respond with its result when auth predicate passes" in new Setup {
+      when(mockAuthConnector.authorise(meq(mockPredicate), meq(EmptyRetrieval))(any(), any()))
+        .thenReturn(Future.successful(()))
```
Comment (arkowalski) [inline test/uk/gov/hmrc/taxhistory/services/SaAuthServiceSpec.scala]: not sure about this one, but should this be a .thenReturn(200)?, so just check

**Pull request merged**